frontend/public/components/cluster-settings/cluster-settings.tsx

```diff
@@ -46,69 +66,115 @@ const getClusterUpdateStatus = (cv: K8sResourceKind): ClusterUpdateStatus => {
+  </React.Fragment>;
+};
+
+const UpdateStatus: React.SFC<UpdateStatusProps> = ({cv, updatesAvailable}) => {
+  const status = updatesAvailable ? ClusterUpdateStatus.UpdatesAvailable : getClusterUpdateStatus(cv);
   const iconClass = getIconClass(status);
   return <React.Fragment>
-    <i className={iconClass} aria-hidden="true" />&nbsp;{status}
+    {
+      updatesAvailable || status === ClusterUpdateStatus.UpdatesAvailable
+        ? <Button bsStyle="link" onClick={()=>(launchUpdateModal(cv))}>
+          <i className={iconClass} aria-hidden={true}></i>
+          &nbsp;
+          {status}
+        </Button>
+        : <span>
+          <i className={iconClass} aria-hidden={true}></i>
+          &nbsp;
+          {status}
+        </span>
+    }
   </React.Fragment>;
 };
 
 const CurrentVersion: React.SFC<CurrentVersionProps> = ({cv}) => {
-  const currentVersion = _.get(cv, 'status.current.version');
+  const currentVersion = getCurrentVersion(cv);
   return currentVersion || <React.Fragment><i className="pficon pficon-warning-triangle-o" aria-hidden="true" />&nbsp;Unknown</React.Fragment>;
 };
 
-const ClusterVersionDetailsTable: React.SFC<ClusterVersionDetailsTableProps> = ({obj: cv}) => {
-  const conditions = _.get(cv, 'status.conditions', []);
-  const retrievedUpdatesFailedCondition = _.find(conditions, { type: 'RetrievedUpdates', status: 'False' });
-  const isFailingCondition = _.find(conditions, { type: 'Failing', status: 'True' });
-  const payload = _.get(cv, 'status.current.payload');
-  return <div className="co-m-pane__body">
-    <div className="co-m-pane__body-group">
-      {isFailingCondition && <FailedConditionAlert message="Upgrade is failing." condition={isFailingCondition} />}
-      {retrievedUpdatesFailedCondition && <FailedConditionAlert message="Could not retrieve updates." condition={retrievedUpdatesFailedCondition} />}
-      <div className="co-detail-table">
-        <div className="co-detail-table__row row">
-          <div className="co-detail-table__section">
-            <dl className="co-m-pane__details">
-              <dt className="co-detail-table__section-header">Channel</dt>
-              <dd>{cv.spec.channel}</dd>
-            </dl>
-          </div>
-          <div className="co-detail-table__section">
-            <dl className="co-m-pane__details">
-              <dt className="co-detail-table__section-header">Update Status</dt>
-              <dd><UpdateStatus cv={cv} /></dd>
-            </dl>
-          </div>
-          <div className="co-detail-table__section">
-            <dl className="co-m-pane__details">
-              <dt className="co-detail-table__section-header">Current Version</dt>
-              <dd><CurrentVersion cv={cv} /></dd>
-            </dl>
+
+
+const ClusterVersionDetailsTable = connectToFlags(FLAGS.CLUSTER_UPDATES_AVAILABLE)(
```

Comment:
If we have the cluster version object, do we need to connect to flags? We should be able to just read the updates from cluster version status.

Reply:
I believe you can do

```suggestion
const ClusterVersionDetailsTable = connectToFlags< ClusterVersionDetailsTableProps>(FLAGS.CLUSTER_UPDATES_AVAILABLE)(
```

Then you shouldn't need to export / use `WithFlagsProps`. (Not tested.)

Reply:
I tried this, but typescript complains that the `flags` prop is missing.

Reply:
Sorry, you're right. I think we could make that work, but what you have looks fine.

Reply:
Changing the way this works. We don't need the flag at all here since we can just use the local storage value for the mock. I'm just going to make the `getClusterUpdateStatus` return `ClusterUpdateStatus.UpdatesAvailable` if the MOCK_CLUSTER_UPDATE value is set in local storage.